File: Python/0009.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # Handle negative numbers
        if x < 0:
            return False
        
        # Convert integer to string
        s = str(x)
        
        # Initialize pointers
        left, right = 0, len(s) - 1
        
        # Check palindrome condition
        while left < right:
            if s[left] != s[right]:
                return False
            left += 1
            right -= 1
        
        return True

File: Scala/0009.scala
object Solution {
  def isPalindrome(x: Int): Boolean = {
    // Handle edge cases
    if (x < 0 || (x % 10 == 0 && x != 0)) return false
    
    var original = x
    var reversed = 0
    
    while (original > reversed) {
      reversed = reversed * 10 + (original % 10)
      original /= 10
    }
    
    // For numbers with odd length, `original` will be equal to `reversed / 10`
    original == reversed || original == reversed / 10
  }
}
